Resteasy Reactive: PatternSyntaxException named capturing group is missing trailing '>'

### Describe the bug

Resteasy Reactive fails with the following exception if `@Path` contains a regex whose group name contains `_`.

Note: Resteasy Classic works fine.

### Expected behavior

_No response_

### Actual behavior

```bash
2021-09-12 10:59:33,807 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:98)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ExceptionInInitializerError
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.lang.Class.newInstance(Class.java:584)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:65)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:119)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
	... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:417)
	... 15 more
Caused by: java.util.regex.PatternSyntaxException: named capturing group is missing trailing '>' near index 4
(?<x_y>[0-9]])$
    ^
	at java.base/java.util.regex.Pattern.error(Pattern.java:2027)
	at java.base/java.util.regex.Pattern.groupname(Pattern.java:2949)
	at java.base/java.util.regex.Pattern.group0(Pattern.java:2995)
	at java.base/java.util.regex.Pattern.sequence(Pattern.java:2123)
	at java.base/java.util.regex.Pattern.expr(Pattern.java:2068)
	at java.base/java.util.regex.Pattern.compile(Pattern.java:1782)
	at java.base/java.util.regex.Pattern.<init>(Pattern.java:1428)
	at java.base/java.util.regex.Pattern.compile(Pattern.java:1068)
	at org.jboss.resteasy.reactive.server.mapping.URITemplate.<init>(URITemplate.java:157)
	at org.jboss.resteasy.reactive.server.core.startup.RuntimeResourceDeployment.buildResourceMethod(RuntimeResourceDeployment.java:126)
	at org.jboss.resteasy.reactive.server.core.startup.RuntimeDeploymentManager.deploy(RuntimeDeploymentManager.java:109)
	at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder.createDeployment(ResteasyReactiveRecorder.java:116)
	at io.quarkus.deployment.steps.ResteasyReactiveProcessor$setupEndpoints-783580934.deploy_1(ResteasyReactiveProcessor$setupEndpoints-783580934.zig:5889)
	at io.quarkus.deployment.steps.ResteasyReactiveProcessor$setupEndpoints-783580934.deploy(ResteasyReactiveProcessor$setupEndpoints-783580934.zig:45)
	at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:386)
	... 15 more
```

### How to Reproduce?

1. https://code.quarkus.io/?e=resteasy-reactive&e=kotlin&extension-search=kotlin
2. Add the following Endpoint

```kotlin
    @GET
    @Path("/{x_y: [0-9]]}")
    fun xyMethod(
        @RestPath("x_y") xy: Int
    ): String {
        return "Hello $xy"
    }
```

### Quarkus version or git rev

2.2.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.8.1

--- a/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java
+++ b/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java
@@ -553,8 +553,8 @@ public abstract class EndpointIndexer<T extends EndpointIndexer<T, PARAM, METHOD
             }
             return method;
         } catch (Exception e) {
-            throw new RuntimeException("Failed to process method " + currentMethodInfo.declaringClass().name() + "#"
-                    + currentMethodInfo, e);
+            throw new RuntimeException("Failed to process method '" + currentMethodInfo.declaringClass().name() + "#"
+                    + currentMethodInfo.name() + "'", e);
         }
     }
 

--- a/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/ServerEndpointIndexer.java
+++ b/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/ServerEndpointIndexer.java
@@ -21,6 +21,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.function.Supplier;
+import java.util.regex.PatternSyntaxException;
 import javax.ws.rs.core.MultivaluedMap;
 import org.jboss.jandex.AnnotationInstance;
 import org.jboss.jandex.ClassInfo;
@@ -45,6 +46,7 @@ import org.jboss.resteasy.reactive.server.core.parameters.converters.ParameterCo
 import org.jboss.resteasy.reactive.server.core.parameters.converters.PathSegmentParamConverter;
 import org.jboss.resteasy.reactive.server.core.parameters.converters.SetConverter;
 import org.jboss.resteasy.reactive.server.core.parameters.converters.SortedSetConverter;
+import org.jboss.resteasy.reactive.server.mapping.URITemplate;
 import org.jboss.resteasy.reactive.server.model.HandlerChainCustomizer;
 import org.jboss.resteasy.reactive.server.model.ServerMethodParameter;
 import org.jboss.resteasy.reactive.server.model.ServerResourceMethod;
@@ -173,6 +175,18 @@ public class ServerEndpointIndexer
             methodAnnotationNames.add(instance.name().toString());
         }
         method.setMethodAnnotationNames(methodAnnotationNames);
+
+        // validate the path
+        validateMethodPath(method, currentClassInfo, info);
+    }
+
+    private void validateMethodPath(ServerResourceMethod method, ClassInfo currentClassInfo, MethodInfo info) {
+        try {
+            new URITemplate(method.getPath(), false);
+        } catch (PatternSyntaxException e) {
+            throw new IllegalArgumentException("Path '" + method.getPath() + "' of method '" + currentClassInfo.name() + "#"
+                    + info.name() + "' is not a valid expression", e);
+        }
     }
 
     protected InjectableBean scanInjectableBean(ClassInfo currentClassInfo, ClassInfo actualEndpointInfo,

